ecs/src/nox/ecs/component/Sprite.cpp

```diff
@@ -0,0 +1,217 @@
+    this->emissiveLight = componentJsonObject.get("emissiveLight", 0.0f).asFloat();
+
+    this->onColorChange(this->color);
+    this->onLightMultiplierChange(this->lightMultiplier);
+    this->onEmissiveLightChange(this->emissiveLight);
+
+    this->renderLevel = componentJsonObject.get("renderLevel", 0).asUInt();
+    this->spriteName = componentJsonObject.get("spriteName", "").asString();
+    this->center = util::parseJsonVec(componentJsonObject["center"], glm::vec2(0.5f));
+    
+    this->spriteRenderNode->setRenderLevel(this->renderLevel);
+    this->spriteRenderNode->addSprite(this->spriteName);
+    this->spriteRenderNode->setActiveSprite(this->spriteName);
+    this->spriteRenderNode->setCenter(this->center);
+
+    onCreate();
+}
+
+void 
+nox::ecs::Sprite::serialize(Json::Value &componentObject)
+{
+    componentObject["scale"] = util::writeJsonVec(this->scale);
+    componentObject["rotation"] = this->rotation;
+    componentObject["flipScale"] = util::writeJsonVec(this->flipScale);
+
+    componentObject["color"] = Json::objectValue;
+    componentObject["color"] = util::writeJsonVec(this->color, util::JsonVecCoordType::COLOR);
+
+    componentObject["spriteName"] = this->spriteName;
+    componentObject["renderLevel"] = this->renderLevel;
+
+    componentObject["center"] = util::writeJsonVec(this->center);
+}
+
+void 
+nox::ecs::Sprite::deactivate()
+{
+    if (this->renderingEnabled == true)
+    {
+        this->broadcastSceneNodeRemoval();
+    }
+}
+
+void 
+nox::ecs::Sprite::activate()
+{
+    if (this->renderingEnabled == true)
+    {
+        this->broadcastSceneNodeCreation();
+    }
+}
+
+void 
+nox::ecs::Sprite::receiveEntityEvent(const ecs::Event& event)
+{
+    if (event.getType() == TypeIdentifier("TransformChange"))
+    {
+        this->updateEntityTransform();
+    }
+}
+
+void 
+nox::ecs::Sprite::setCenter(const glm::vec2& center)
+{
+    this->center = center;
+    this->spriteRenderNode->setCenter(this->center);
+}
+
+void 
+nox::ecs::Sprite::setActiveSprite(const std::string& spriteName)
+{
+    this->spriteRenderNode->setActiveSprite(spriteName);
+}
+
+void 
+nox::ecs::Sprite::setRenderingState(const bool enable)
+{
+    if (enable == true && this->renderingEnabled == false)
+    {
+        this->broadcastSceneNodeCreation();
+        this->renderingEnabled = true;
+    }
+
+    if (enable == false && this->renderingEnabled == true)
+    {
+        this->broadcastSceneNodeRemoval();
+        this->renderingEnabled = false;
+    }
+}
+
+void 
+nox::ecs::Sprite::setRenderLevel(unsigned int renderLevel)
+{
+    if (this->renderLevel != renderLevel)
+    {
+        this->renderLevel = renderLevel;
+        this->spriteRenderNode->setRenderLevel(renderLevel);
+    }
+}
+
+void 
+nox::ecs::Sprite::addSprite(const std::string& spriteName)
+{
+    this->spriteRenderNode->addSprite(spriteName);
+}
+
+void 
+nox::ecs::Sprite::onCreate()
+{
+    auto tempHandle = this->entityManager->getComponent(this->id,
```

Comment:
No need to create a temp here, why not just assign it straight to the transformHandle?